CMakeLists.txt

```diff
@@ -114,12 +116,18 @@ target_compile_options(nodes
   PRIVATE
     $<$<CXX_COMPILER_ID:MSVC>:/W4 /wd4127 /EHsc>
     $<$<CXX_COMPILER_ID:GNU>:-Wall -Wextra>
-    $<$<CXX_COMPILER_ID:Clang>:-Wall -Wextra>
 )
+if(NOT "${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC")
+  # Clang-Cl on MSVC identifies as "Clang" but behaves more like MSVC:
+  target_compile_options(nodes
+    PRIVATE
+      $<$<CXX_COMPILER_ID:Clang>:-Wall -Wextra>
+  )
+endif()
 
 target_compile_features(nodes
   PUBLIC
-    cxx_generic_lambdas # Require C++14
+    cxx_generic_lambdas # Require C++14 or newer
```

Comment:
In the resent versions CMake has added the possibility to write the needed standard directly.
I tend to use this expression:

```
target_compile_features(mylib PUBLIC cxx_std_14)
```